src/bloqade/ir/location/transform.py

```diff
@@ -164,7 +271,51 @@ def apply_defect_density(
         defect_probability: float,
         rng: np.random.Generator = np.random.default_rng(),
     ):
-        """apply defect_probability randomly to existing atom arrangement."""
+        """
+        Drop atoms randomly with `defect_probability` probability (range of 0 to 1).
+        Internally this occurs by setting certain sites to have a SiteFilling
+        set to false indicating no atom is present at the coordinate.
+
+        A default numpy-based Random Number Generator is used but you can
+        explicitly override this by passing in your own.
+
+        Usage Example:
```

Comment:
if we are doing markdown, shouldn't these be level-3 sections?

Reply:
Could you elaborate on what you mean by level-3 sections? 

I'll fix anything that breaks markdown compatibility 👍

Reply:
I mean 

```
### Examples
```

since that gives you the section title? otherwise it's in ReText style?